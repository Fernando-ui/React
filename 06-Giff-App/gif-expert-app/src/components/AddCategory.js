import React, { useState } from 'react'

export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('Hola mundo');
    const handleInputChange = (e) => {

        setInputValue(e.target.value);
    }

    const handleSubmit = (e) => {

        e.preventDefault();
        console.log('haciendo el submit');
        setCategories( cats => [ ... cats, inputValue]);

    }

    return (
        <>
        <form onSubmit = { handleSubmit}>

            <input 
            type="text"
            value={ inputValue }
            onChange = { (e) => {handleInputChange(e)}}

            />

        </form>
        </>
    )
}
