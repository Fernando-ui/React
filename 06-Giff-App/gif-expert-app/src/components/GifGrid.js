import React, { useState, useEffect } from 'react'
import { GridGifItem } from './GridGifItem';

export const GifGrid = ({ category}) => {

    const [images, setImages] = useState([]);

    useEffect( ()=> {

        getGifs();
    },[]);


    const getGifs = async()=>{


        const url = 'https://api.giphy.com/v1/gifs/search?q=Rick+and+morty&limit=10&api_key=N8Cf1CkzsWIo20asLCPDWM5B69xLgQzR';
        const resp = await fetch(url);
        // console.log(resp);
        const {data} = await resp.json();
        // console.log(data,'Toda la data');
        const gifs = data.map(img => {

            return{

                id:img.id,
                title:img.title,
                url:img.images.downsized_medium.url,
            }
        })

        console.log(gifs);
        setImages(gifs);
        
    }

    // getGifs();
    return (
        <>
            <h3>{category}</h3>
       
            {
                images.map( (img) => (

                    <GridGifItem 
                    key={img.id}
                    {...img}
                    />
                ))
            }
        
        
        </>
    )
}
