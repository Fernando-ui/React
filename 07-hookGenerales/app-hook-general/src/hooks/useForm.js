import  { useState } from 'react'

export const useForm = ( initialState = {}) => {
    
    const [ values, setValues ] = useState(initialState);

    const reset = () => {

        setValues(initialState);

    }

    // * Este tiene el target prque se le envia el evento y del evento se desesctructura el target ya con el es con el que trabajamos 
    // * Lo que cambio es lo que estamos capturando, en este caso seria lo que escribimos en el input
    const handleInputChange = ({ target }) => {

        setValues({ 

            ...values,
            [target.name]:target.value
        })
    }

    return[values, handleInputChange, reset];
}
