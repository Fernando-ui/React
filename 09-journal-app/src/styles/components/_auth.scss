.auth__main{

    align-items: center;
    background-color: $primary;
    display:flex;
    justify-content: center;
    margin:0px;
    height:100vh;
    width:100vw;
}

.auth__box-container{

    background-color:#fff;
    box-shadow:0px 3px $dark-grey;
    border-radius:15px;
    padding:20px;
    width:50%;
}

.auth__title{

    color:darken($primary,20%);
    margin-bottom:20px;
}

.auth__input{
    color:$dark-grey;
    border:0px;
    border-bottom:1px solid $light-grey;
    font-size:16px;
    margin-bottom:20px;
    height:20px;
    width:100%;

    transition: border-bottom .3s ease;

    &:focus{
        outline:0;
        border-bottom: 1px solid #6A6F73;
    }
}

.auth__social-networks{

    align-items:center;
    display:flex;
    justify-content:center;
    flex-direction: column;
    padding-top:20px;
    padding-bottom:20px;
    width:100%;

}

.auth__alert-error{

    background-color:red;
    border-radius:5px;
    color:#fff;
    display:flex;
    justify-content:center;
    margin-bottom:10px;
    padding:5px;
}
