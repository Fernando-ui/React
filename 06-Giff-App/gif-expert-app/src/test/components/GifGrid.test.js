import '@testing-library/jest-dom'
import { GifGrid } from '../../components/GifGrid';
import { shallow } from 'enzyme';
import { useFetchGifs } from '../../hooks/useFetchGifs';
import React from 'react'


jest.mock('../../hooks/useFetchGifs'); 

describe('Pruebas con <GifGrid />', () => {

    const categoria = 'Rick and Morty';
    
    test('Debe de coincidir el snapshot ', () => {
        
        useFetchGifs.mockReturnValue({
            
            data:[],
            loading:true
            
        });
        
        const wrapper = shallow(<GifGrid category={categoria} />);
        expect(wrapper).toMatchSnapshot();

    });
    
    test('Debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {

        const img = [{
            id:'abc', 
            url:'https://algo.jpg',
            title:'cualquier cosa'
        }]

        useFetchGifs.mockReturnValue({
            
            data:[img],
            loading:false
            
        });
        
        const wrapper = shallow(<GifGrid category={ categoria} />);        

        expect(wrapper.find('p').exists()).toBe(false)
        expect(wrapper.find('GifGridItem').length).toBe(img.length)
    });
    
    test('Debe hacer match con el snapShoot', () => {
    
        

    });
    

})
