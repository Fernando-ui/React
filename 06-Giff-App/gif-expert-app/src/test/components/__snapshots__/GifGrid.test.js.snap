// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con <GifGrid /> Debe de coincidir el snapshot  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    Rick and Morty
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
